### PR TITLE
editor: record revisions api

### PR DESCRIPTION
**Pull request by harunurhan**
* Adds views for getting revisions and reverting.

### CODE REVIEW

File: inspirehep/modules/editor/views.py [hunk 1]
```diff
@@ -42,6 +51,47 @@
...
 
 
+@blueprint.route('/revisions/<rec_uuid>/revert', methods=['PUT'])
+def revert_to_revision(rec_uuid):
+    """Revert given record to given revision"""
+    record = Record.get_record(rec_uuid)
+    revision_id = request.json['revision_id']
+    record.revert(revision_id)
+    db.session.commit()
+    return jsonify(succes=True)
```
Comment (jmartinm) [inline inspirehep/modules/editor/views.py]: Typo ;)

File: inspirehep/modules/editor/views.py [hunk 2]
```diff
@@ -42,6 +51,56 @@
...
+
+    pid = PersistentIdentifier.query.filter_by(pid_value=recid).one()
+    record = get_db_record(pid.pid_type, recid)
+
+    revisions = []
+    for revision in record.revisions:
+        transaction_id = revision.model.transaction_id
+
+        user_email = db.session.query(Transaction).filter(
+            Transaction.id == transaction_id).one().user.email
```
Comment (harunurhan) [inline inspirehep/modules/editor/views.py]: can we have a transaction without a user? like first db insert! In that case that would crash.
@jmartinm

File: inspirehep/modules/editor/views.py [hunk 3]
```diff
@@ -42,6 +51,56 @@
...
+
+    pid = PersistentIdentifier.query.filter_by(pid_value=recid).one()
+    record = get_db_record(pid.pid_type, recid)
+
+    revisions = []
+    for revision in record.revisions:
+        transaction_id = revision.model.transaction_id
+
+        user_email = Transaction.query.filter(
+            Transaction.id == transaction_id).one().user.email
```
Comment (harunurhan) [inline inspirehep/modules/editor/views.py]: can we have a transaction without a user? like first db insert! In that case that would crash.
@jmartinm

File: inspirehep/modules/editor/views.py [hunk 4]
```diff
@@ -42,6 +51,59 @@
                       url_prefix='/editor',)
 
 
+@blueprint.route('/revisions/<rec_uuid>/revert', methods=['PUT'])
```
Comment (harunurhan) [inline inspirehep/modules/editor/views.py]: using `rec_uuid` here and `recid` in another route makes it a bit weird but avoids

```
pid = PersistentIdentifier.query.filter_by(pid_value=recid).one()
record = get_db_record(pid.pid_type, recid)
```

and uses more simple

```record = Record.get_record(rec_uuid)```

Comment (harunurhan) [reply inspirehep/modules/editor/views.py]: but also because of this, we have to return, `rec_uuid` from `get_revisions` method.

**Comment by harunurhan** [comment]
@jacquerie for the test I am actually not sure what to test, here except `'/revisions/<recid>'`'s logic with `for` and `if`s.

**Comment by jacquerie** [comment]
> @jacquerie for the test I am actually not sure what to test

What about the authentication, like @jmartinm suggested?

**Comment by jmartinm** [comment]
> What about the authentication, like @jmartinm suggested?

Once we merge https://github.com/inspirehep/inspire-next/pull/2724 I will create a PR with the permissions for the editor and then we can rebase this PR on top.

File: tests/integration/editor/test_editor_views.py [hunk 5]
```diff
@@ -287,7 +384,8 @@ def test_refextract_text(log_in_as_cataloger, api_client):
 
     assert response.status_code == 200
     assert validate(references, subschema) is None
-    assert get_value({'references': references}, 'references.reference.publication_info.journal_title')
+    assert get_value({'references': references},
+                     'references.reference.publication_info.journal_title')
```
Comment (jacquerie) [inline tests/integration/editor/test_editor_views.py]: Check your editor! Our `pytest.ini` ignores `E501`, so your style checks should be ignoring it as well.

**Comment by harunurhan** [comment]
tests are passing when I rebased to #2830, so after it is merged this can be merged finally.

@jacquerie you can have a loot at the tests, if they are OK.

**Comment by jacquerie** [comment]
@harunurhan @jmartinm in general, you should sign your commits.

If you don't want to remember adding `-s` you can add it to the configuration:
```shell
$ git config --local format.signoff true
```
(It's even part of the documentation! http://inspirehep.readthedocs.io/en/latest/dev_flow.html.)

**Comment by jmartinm** [comment]
@harunurhan if you are amending the commits, I would also squash into one commit so that the functionality and the tests are in the same commit.

**Pull request merged**